### PR TITLE
Enable listing cart on explore items and collection pages

### LINKED ISSUE #7355: Select on collection view from owned items but can't do mass list
On collection view, I select card and do nothing.
- https://canary.kodadot.xyz/ahk/collection/132?owned=true
image

I would expect to be able to have toast on the bottom as on the profile

image

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context
- [x] branch from #7333
- [x] Closes #7355 
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?
yes **$**

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Screenshot 📸

- [x] My fix has changed UI
image
image

### <samp>🤖 Generated by Copilot at 119ff1d</samp>

This pull request adds a listing cart feature that allows users to list multiple NFTs for sale or buy multiple NFTs at once. It modifies the `ItemsGrid`, `ProfileDetail`, and `explore-layout` components to integrate the feature, and adds new files and functions to handle the data fetching and configuration logic. It also refactors some existing code to improve performance and readability.

### <samp>🤖 Generated by Copilot at 119ff1d</samp>

> _We are the masters of the listing cart_
> _We can sell our NFTs with a single click_
> _We don't need no `ItemsGridImage` logic_
> _We have the power of the `useItemsGrid` trick_

### CODE REVIEW

**Comment by daiagi** [comment]
best to merge #7333 before reviewing this one
drafting it until then

**Comment by yangwao** [comment]
Not related with this pr
That amount of zeroes after decimal, is it needed ? 
image

File: components/items/ItemsGrid/ItemsGrid.vue [hunk 1]
```diff
@@ -149,11 +151,16 @@ watch(
...
 )
 
-onBeforeMount(async () => {
-  await fetchSearch({
+onBeforeMount(() => {
+  if (listingCartEnabled.value) {
+    listingCartStore.clear()
+  }
+
+  fetchSearch({
```
Comment (roiLeo) [inline components/items/ItemsGrid/ItemsGrid.vue:159]: why remove async/await?

File: composables/useListingCartConfig.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export const displayOnPages = [
```
Comment (roiLeo) [inline composables/useListingCartConfig.ts:1]: why create a new composable instead of puting the code inside cart `store`?

**Review by roiLeo — APPROVED** [review]
otherwise lg

File: composables/useListingCartConfig.ts (hunk 2 above)
Comment (daiagi) [reply composables/useListingCartConfig.ts:1]: It made sense to me to have config out of the store

Let me know if you disagree and I'll change that

File: components/items/ItemsGrid/ItemsGrid.vue (hunk 1 above)
Comment (roiLeo) [reply components/items/ItemsGrid/ItemsGrid.vue:159]: alright

**Comment by yangwao** [comment]
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 4.05 USD/DOT ~ 7.407 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x74bdf3e66e60d4b175ebcb860cd5ac15c906ea6d9172c3ca71a15ac50f6f92d5](https://polkadot.subscan.io/extrinsic/0x74bdf3e66e60d4b175ebcb860cd5ac15c906ea6d9172c3ca71a15ac50f6f92d5)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**